pageserver/src/bin/pageserver.rs

```diff
@@ -87,7 +87,7 @@ fn main() -> anyhow::Result<()> {
     };
 
     // Initialize logging, which must be initialized before the custom panic hook is installed.
-    logging::init(conf.log_format)?;
+    logging::init_with_span_assert(conf.log_format, Some(pageserver::SPAN_ASSERTIONS.clone()))?;
```

Comment:
The `SPAN_ASSERTIONS` should not be enabled for release builds.
At least not by default, they have considerable overhead.

Also need to look at increase in test runtime to figure out overhead in debug builds.